src/common/raft/integration/cagra_instantiations.cu

```diff
@@ -0,0 +1,27 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2023,NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#undef RAFT_EXPLICIT_INSTANTIATE_ONLY
+#include "common/raft/integration/type_mappers.hpp"
+#include "common/raft/proto/filtered_search_instantiation.cuh"
+#include "common/raft/proto/raft_index_kind.hpp"
+
+RAFT_FILTERED_SEARCH_INSTANTIATION(cagra, raft_knowhere::raft_data_t<raft_proto::raft_index_kind::cagra>,
```

Comment:
Here we actually instantiate the extern templates we use in `cagra_index.cu`. Note that we only include a very limited set of headers here, so changes in knowhere should only rarely trigger a rebuild of the underlying algorithm.